### PR TITLE
Implement new beta CI/CD

### PR DESCRIPTION
**Pull request by Renetnielsen**
## References

* #0000

### CODE REVIEW

File: .github/workflows/detect-changes.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    # For pull requests it's not necessary to checkout the code because GitHub REST API is used to determine changes
+    - name: Checkout repository
+      if: ${{ github.event_name != 'pull_request' }}
+      uses: actions/checkout@v3
+
+    - name: Detect file changes
+      uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
```
Comment (dstenroejl) [inline .github/workflows/detect-changes.yml:49]: Both filters should also include:
            - '.github/workflows/cd.yml'
            - '.github/workflows/detect-changes.yml'

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -19,30 +19,82 @@ on:
...
+  #
 
-  update_coverage_report:
+  dotnet_update_coverage_report:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
     uses: Energinet-DataHub/.github/.github/workflows/dotnet-solution-ci.yml@v8
     with:
       SOLUTION_FILE_PATH: 'source/Energinet.DataHub.PostOffice.sln'
-      USE_COSMOS_DB_EMULATOR: true
```
Comment (djorgensendk) [inline .github/workflows/cd.yml:41]: has it been verified that USE_COSMOS_DB_EMULATOR can be removed?

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -19,30 +19,82 @@ on:
...
+      - name: Find associated pull request
+        uses: Energinet-DataHub/.github/.github/actions/find-related-pr-number@v8
+        id: find_pull_request
+
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.PAT_TOKEN }}
+          repository: '${{ secrets.ENVIRONMENT_REPOSITORY_PATH }}'
+          event-type: 'geh-timeseries-deployment-request-domain'
```
Comment (djorgensendk) [inline .github/workflows/cd.yml:99]:
```suggestion
          event-type: 'geh-post-office-deployment-request-domain'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -20,39 +20,92 @@ on:
...
+  dotnet_ci_build:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
+    uses: Energinet-DataHub/.github/.github/workflows/dotnet-build-prerelease.yml@v8
     with:
       SOLUTION_FILE_PATH: 'source/Energinet.DataHub.PostOffice.sln'
-      USE_COSMOS_DB_EMULATOR: true
+
+  # Run all tests in 'Tests.dll'
+  dotnet_ci_test_unittests:
```
Comment (dstenroejl) [inline .github/workflows/ci.yml:63]: Not important, but maybe we should try aligning the job name for tests. In some repos we use "dotnet_ci_test_Xxx" in others we just call them "unit_tests", "integration_tests". Here we use a combination of those (below we have "integration_test"). Maybe choose one format and use that for both (here and line 73).

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -20,39 +20,92 @@ on:
...
-      CALLER_REPOSITORY_PATH: Energinet-DataHub/geh-post-office
-      RELEASE_NAME_PREFIX: dotnet
-    secrets:
-      PAT_TOKEN: ${{ secrets.PAT_TOKEN }}
+    if: |
+      always() &&
+      !contains(needs.*.result, 'failure') &&
+      !contains(needs.*.result, 'cancelled')
+    steps:
+      - run: echo "${{ toJSON(needs) }}"git git
```
Comment (djorgensendk) [inline .github/workflows/ci.yml:111]: why git git?

File: .github/workflows/detect-changes.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          infrastructure:
+            - '.github/workflows/ci.yml'
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:51]: also add: cd.yml and the detech-changes.yml

File: .github/workflows/detect-changes.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            - 'build/**'
+          dotnet:
+            - '.github/workflows/ci.yml'
+            - '.github/actions/dotnet-*/**'
```
Comment (djorgensendk) [inline .github/workflows/detect-changes.yml:55]: also add: cd.yml and the detech-changes.yml

**Review by djorgensendk — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 5 above)
Comment (Renetnielsen) [reply .github/workflows/ci.yml:111]: Haha my bad, probably thought i was writing in the terminal. Will remove it

Comment (dstenroejl) [inline .github/workflows/ci.yml:111]: Why "git git"? 👯

File: .github/workflows/cd.yml (hunk 2 above)
Comment (dstenroejl) [inline .github/workflows/cd.yml:41]: The parameters here should match CI - do they not use Cosmos DB?

**Review by dstenroejl — COMMENTED** [review]
Added comments

**Review by dstenroejl — APPROVED** [review]

**Review by djorgensendk — APPROVED** [review]

**Review by dstenroejl — APPROVED** [review]

**Pull request merged**